tests/functional/h/views/admin/test_permissions.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+
+class TestAdminPermissions:
+    PAGES = (
+        # URL, accessible by role.Staff?
+        ("/admin/", True),
+        ("/admin/admins", False),
+        ("/admin/badge", False),
+        ("/admin/features", False),
+        ("/admin/groups", True),
+        ("/admin/mailer", True),
+        ("/admin/nipsa", False),
+        ("/admin/oauthclients", False),
+        ("/admin/organizations", True),
+        ("/admin/staff", False),
+        ("/admin/users", True),
+        ("/admin/search", False),
+    )
+
+    @pytest.mark.parametrize("url", (page[0] for page in PAGES))
```

Comment:
Doing these one by one causes these tests to be very slow, but it's the most `pytest`'y unit testy way to do it. This is due to the DB tear down between each thing.

It's _much_ faster to do the setup and login the user once, then just scan over the entire list. I'm not sure if that's ok or not, but it's not really a sustainable solution as I suspect other permissions tests would not be as amenable to that style.

Instead I've added a different PR to attempt to speed up all tests by removing the DB tear down where not necessary: https://github.com/hypothesis/h/pull/6845